src/plugins/gmx/gmxplugin.cpp

```diff
@@ -106,34 +105,31 @@ bool GmxPlugin::write(const Map *map, const QString &fileName)
 
     uint tileId = 0u;
     int currentLayer = map->layers().size();
-
-    foreach (const Layer *layer, map->layers()) {
+    for (const Layer *layer : map->layers()) {
 
         if (layer->layerType() != Layer::TileLayerType) continue;
 
+        QString depth = QString::number(layer->hasProperty(QLatin1String("Depth")) ? layer->property(QLatin1String("Depth")).toInt() : currentLayer);
+
         const TileLayer *tileLayer = static_cast<const TileLayer*>(layer);
         for (int y = 0; y < tileLayer->height(); ++y) {
             for (int x = 0; x < tileLayer->width(); ++x) {
                 const Cell &cell = tileLayer->cellAt(x, y);
                 const Tile *tile = cell.tile;
-
                 if(tile) {
                     stream.writeStartElement("tile");
 
                     stream.writeAttribute("bgName", tile->tileset()->name());
-                    stream.writeAttribute("x", QString::number((int)(x * map->tileWidth())));
-                    stream.writeAttribute("y", QString::number((int)(y * map->tileHeight())));
-                    stream.writeAttribute("w", QString::number(map->tileWidth()));
-                    stream.writeAttribute("h", QString::number(map->tileHeight()));
+                    stream.writeAttribute("x", QString::number((int)(x * tile->width())));
+                    stream.writeAttribute("y", QString::number((int)(y * tile->height())));
```

Comment:
When I asked you to use the tile size, I meant just for `w` and `h`. The position of the tile depends on the map tile size, not on the size of each tile.